src/transformers/utils/quantizers.py

```diff
@@ -0,0 +1,715 @@
+                    config_origin = "config"
+
+            else:
+                quantization_method = config.quantization_config.get("quant_method", QuantizationMethod.BITS_AND_BYTES)
+                quantization_config = config.quantization_config
+                config_origin = "config"
+
+        elif self.quantization_method_from_args is not None:
+            quantization_config = self.quantization_config_from_args
+            quantization_method = self.quantization_method_from_args
+            config_origin = "args"
+
+        else:
+            # Normal loading without quantization. No quant arguments present in args or model config.
+            return
+
+        # set quantizer and config classes ========================================================================
+        if quantization_method == QuantizationMethod.BITS_AND_BYTES:
+            config_class = BitsAndBytesConfig
+            if dict(quantization_config).get("load_in_4bit", False):
+                quantizer_class = BnbFourBitHFQuantizer
+            elif dict(quantization_config).get("load_in_8bit", False):
+                quantizer_class = BnbEightBitHFQuantizer
+            else:
+                raise ValueError("BnB config should specify either `load_in_4bit` or `load_in_8bit`")
+
+        elif quantization_method == QuantizationMethod.GPTQ:
+            quantizer_class = GPTQHFQuantizer
+            config_class = GPTQConfig
+
+        else:
+            # unknown quantization_method, which is not None:
+            raise NotImplementedError(
+                f"Unsupported quantization method detected: {quantization_method}. Check for updates."
+            )
+
+        # finishing touches ========================================================
+        if not isinstance(quantization_config, (dict, QuantizationConfigMixin)):
+            raise ValueError(
+                f"Invalid type for `quantization_config`: {type(quantization_config)}. Should be a `dict` or a"
+                " `QuantizationConfigMixin` subclass instance."
+            )
+
+        quantization_config = config_class.from_dict(config_dict=dict(quantization_config))
+
+        config.quantization_config = quantization_config
+        quantizer = quantizer_class(quantization_config=quantization_config, config_origin=config_origin)
+        return quantizer
+
+
+def get_module_from_name(module, tensor_name):
+    if "." in tensor_name:
+        splits = tensor_name.split(".")
+        for split in splits[:-1]:
+            new_module = getattr(module, split)
+            if new_module is None:
+                raise ValueError(f"{module} has no attribute {split}.")
+            module = new_module
+        tensor_name = splits[-1]
+    return module, tensor_name
+
+
+class HFQuantizer(ABC):
+    requires_parameters_quantization = False
+
+    @property
+    @abstractmethod
+    def method(self):
+        """placeholder to method from QuantizationMethod"""
+        pass
```

Comment:
```suggestion
        ...
```
You can just set `...` instead of `pass` I think since it is an abstract method

Reply:
done